cmd/maya-apiserver/app/server/snapshot_endpoint.go

```diff
@@ -0,0 +1,180 @@
+package server
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/openebs/maya/types/v1"
+	"github.com/openebs/maya/volume/provisioners/jiva"
+)
+
+// SnapshotSpecificGetRequest deals with HTTP GET request w.r.t a Volume Snapshot
+func (s *HTTPServer) SnapshotSpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	// Extract info from path after trimming
+	path := strings.TrimPrefix(req.URL.Path, "/latest/snapshot")
+
+	// Is req valid ?
+	if path == req.URL.Path {
+		fmt.Println("Request coming", path)
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	switch {
+	case strings.Contains(path, "/create/"):
+		return s.SnapshotCreate(resp, req)
+	case strings.Contains(path, "/revert/"):
+		return s.SnapshotRevert(resp, req)
+	case strings.Contains(path, "/list"):
+		volName := strings.TrimPrefix(path, "/list/")
+		return s.SnapshotList(resp, req, volName)
+	default:
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+}
+
+// SnapshotCreate is http handler which handles snaphsot-create request
+func (s *HTTPServer) SnapshotCreate(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+
+	if req.Method != "PUT" && req.Method != "POST" {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	fmt.Println("[DEBUG] Processing Volume Snapshot request")
```

Comment:
This should be more than DEBUG level.

Reply:
Can we use glog ?

Reply:
Yes , this was initial commit , i am on it.

Reply:
done